src/ConnectKafka.cpp

```diff
@@ -30,17 +47,25 @@ std::unique_ptr<RdKafka::Metadata> ConnectKafka::queryMetadata() {
   RdKafka::Metadata *metadataRawPtr(nullptr);
   Consumer->metadata(true, nullptr, &metadataRawPtr, 1000);
   std::unique_ptr<RdKafka::Metadata> metadata(metadataRawPtr);
-  if (!metadata) {
-    throw std::runtime_error("Failed to query metadata from broker");
+  try {
+    if (!metadata) {
+      throw std::runtime_error("Failed to query metadata from broker");
+    }
+  } catch (std::runtime_error &E) {
+    Logger->error(E.what());
+  } catch (std::exception &E) {
+    Logger->error(E.what());
   }
   return metadata;
 }
 
-ConnectKafka::ConnectKafka(std::string Broker, std::string ErrStr)
-    : Logger(spdlog::get("LOG")) {
+ConnectKafka::ConnectKafka(std::string Broker) : Logger(spdlog::get("LOG")) {
+  std::string ErrStr;
   this->Consumer =
       std::shared_ptr<RdKafka::KafkaConsumer>(RdKafka::KafkaConsumer::create(
           createGlobalConfiguration(Broker).get(), ErrStr));
+  if (!ErrStr.empty())
+    Logger->error(ErrStr);
```

Comment:
It would be good to add something to the logged error to say where this error occurred ("Error creating Kafka Consumer in ConnectKafka...").